library/src/com/davemorrissey/labs/subscaleview/SubsamplingScaleImageView.java

```diff
@@ -762,14 +762,14 @@ public boolean onTouchEvent(@NonNull MotionEvent event) {
                             fitToBounds(true);
                             if (!isVerticalScrollingParent) {
                                 boolean atXEdge = lastX != vTranslate.x;
-                                // dx*0.5 so it's similar to ViewPager class
-                                boolean edgeXSwipe = atXEdge && dx*0.5 > dy && !isPanning;
-                                // dy > 5 allows for better diagonally panning than > 15
-                                boolean yPan = lastY == vTranslate.y && dy > 5;
+                                // dx*0.25 is more natural for thumb movement
+                                boolean edgeXSwipe = atXEdge && dx*0.20 > dy && !isPanning;
+                                // dy > 1.5 is more natural for quick and small thumb movement
+                                boolean yPan = lastY == vTranslate.y && dy > 1.5;
 
                                 if (!edgeXSwipe && (!atXEdge || yPan || isPanning)) {
                                     isPanning = true;
-                                } else if (dx > 5) {
```

Comment:
I didn't change anything of the original version here. I changed this to .0 so it is more clear that this is indeed **no** integer. 
And to be fair, the whole sequence is quite long, so I doubt comparing integers here would speed up a lot.